### PR TITLE
feat: Adding new docs for sentry-cocoa 5.x

### PR DESCRIPTION
**Pull request by HazAT**
screencapture-0-0-0-0-9001-platforms-cocoa-2020-03-17-11_59_22
0 0 0 0_9001_platforms_cocoa_usage__platform=swift(iPad Pro)

### CODE REVIEW

File: src/collections/_documentation/platforms/cocoa/dsym.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+```
+{%- endcapture -%}
+{%- include components/alert.html
+  title="On Prem"
+  content=__alert_content
+%}
+
+### Using `sentry-cli`
+
+There are two ways to download the dSYM from iTunesConnect. After that, you can upload the dSYM using [sentry-cli](https://github.com/getsentry/sentry-cli/releases).
```
Comment (PeloWriter) [inline src/collections/_documentation/platforms/cocoa/dsym.md]: I find this a bit confusing. We mention two ways to download, but I'm not sure what the methods are? 

For structuring, suggest: "Download the dSYM from iTunesConnect using either ... or ..."

File: src/collections/_documentation/platforms/cocoa/index.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
+---
+title: Cocoa
+---
+
+{% capture __alert_content -%}
+Version `5.0.0` is still in beta.
```
Comment (PeloWriter) [inline src/collections/_documentation/platforms/cocoa/index.md]: remove "still"

File: src/collections/_documentation/platforms/cocoa/index.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+<!-- WIZARD -->
+## Installation {#install}
+
+The SDK can be installed using [CocoaPods](http://cocoapods.org), [Carthage](https://github.com/Carthage/Carthage), or [Swift Package Manager](https://swift.org/package-manager/). This is the recommended SDK for both Swift and Objective-C projects.
+
+It supports iOS (>= 8.0), tvOS (>= 9.0), macOS (>= 10.10) and partially watchOS (>= 2.0). 
+
+### CocoaPods
+
+Installing the SDK with CocoaPods is currently our recommended way.
```
Comment (PeloWriter) [inline src/collections/_documentation/platforms/cocoa/index.md]: Suggest: "We recommend installing the SDK with CocoaPods." because "currently" dates very quickly.

File: src/collections/_documentation/platforms/cocoa/index.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+platform :ios, '8.0'
+use_frameworks! # This is important
+
+target 'YourApp' do
+    pod 'Sentry', :git => 'https://github.com/getsentry/sentry-cocoa.git', :tag => '5.0.0-beta.0'
+end
+```
+<!-- {% sdk_version sentry.cocoa %} -->
+
+Afterwards run `pod install`. In case you encounter problems with dependencies and you are on a newer CocoaPods you might have to run `pod repo update` first.
```
Comment (PeloWriter) [inline src/collections/_documentation/platforms/cocoa/index.md]: Should we recommend running `pod repo update` first, regardless?

File: src/collections/_documentation/platforms/cocoa/index.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+### Swift Package Manager
+
+To integrate Sentry into your Xcode project using SPM, open your App in Xcode and open `File` -> `Swift Packages` -> `Add Package Dependency`. Then add the SDK by entering the git repo url `https://github.com/getsentry/sentry-cocoa.git`, and select a version (or branch) on the next page.
+
+NOTE: Version tags or branches need to have the Package.swift file in it or Xcode won't be able to install the package.
+
+## Configuration {#configure}
+
+Make sure you initalize the SDK as soon as possible in your application lifecycle e.g. in you AppDelegate’s `application:didFinishLaunchingWithOptions` method:
```
Comment (PeloWriter) [inline src/collections/_documentation/platforms/cocoa/index.md]: is this "your AppDelegate's"?

File: src/collections/_documentation/platforms/cocoa/index.md [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+
+{% include components/platform_content.html content_dir='init' %}
+
+## Debug Symbols {#sentry-cocoa-debug-symbols}
+
+Before you can start capturing crashes you will need to tell Sentry about the debug information by uploading dSYM files. Depending on your setup this can be done in different ways:
```
Comment (PeloWriter) [inline src/collections/_documentation/platforms/cocoa/index.md]: Suggest, "Before capturing crashes, you need to provide debug information to Sentry. Debug information is provided by uploading dSYM files using one of two methods, dependent on your setup:"

File: src/collections/_documentation/platforms/cocoa/index.md [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+
+-   [With Bitcode]({%- link _documentation/platforms/cocoa/dsym.md -%}#dsym-with-bitcode)
+-   [Without Bitcode]({%- link _documentation/platforms/cocoa/dsym.md -%}#dsym-without-bitcode)
+
+<!-- ENDWIZARD -->
+
+## Testing if the SDK works
+
+The simplest way to figure out if the SDK is setup correctly and connected to your Sentry project, just simply capture a message:
```
Comment (PeloWriter) [inline src/collections/_documentation/platforms/cocoa/index.md]: Suggest "To validate that the SDK is correctly set up and connected to your Sentry project, capture a test message:"

File: src/collections/_documentation/platforms/cocoa/index.md [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+
+{% include components/platform_content.html content_dir='captureMessage' %}
+
+If everything is working correctly, this message should appear within seconds in your Sentry project.
+
+## Crash Handling
+
+By default our SDK hooks into all the signal and exception handlers (for macOS there is one additional step -> see [Advanced Usage]({%- link _documentation/platforms/cocoa/usage.md -%})).
```
Comment (PeloWriter) [inline src/collections/_documentation/platforms/cocoa/index.md]: Suggest breaking out the MacOS exception more clearly. Perhaps "Our SDK hooks into all signal and exception handlers, except for MacOS. If you are using MacOS, please see the additional step provided in [Advanced Usage]({%- link _documentation/platforms/cocoa/usage.md -%})

File: src/collections/_documentation/platforms/cocoa/index.md [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+If you want to try out if it works for you, the SDK provides a test crash function in order to crash your app:
```
Comment (PeloWriter) [inline src/collections/_documentation/platforms/cocoa/index.md]: Suggest, "To try it out, the SDK provides a test crash function:"

File: src/collections/_documentation/platforms/cocoa/index.md [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+
+{% include components/platform_content.html content_dir='crash' %}
+
+_If you crash with a debugger attached, nothing will happen._
+
+Crashes are only submitted upon re-launching the application. To see the crash in Sentry, close the app and launch it again.
```
Comment (PeloWriter) [inline src/collections/_documentation/platforms/cocoa/index.md]: Suggest: "Crashes are submitted only upon re-launch of the application. To view the crash in Sentry, close your app and re-launch it."

File: src/collections/_documentation/platforms/cocoa/pass-scope/swift.md [hunk 11]
```diff
@@ -0,0 +1,10 @@
+```swift
+let exception = NSException(name: NSExceptionName("My Custom exeption"), reason: "User clicked the button", userInfo: nil)
+let scope = Scope()
+scope.setLevel(.fatal)
+// By explicity just passing the scope, only the data in this scope object will be added to the event
+// The global scope (calls to configureScope) will be ignored
+// Only do this if you know what you are doing, you loose a lot of useful info
```
Comment (PeloWriter) [inline src/collections/_documentation/platforms/cocoa/pass-scope/swift.md]: Do we want to be more explicit here, this comment is hard to parse. Suggest "Only do this if you have mastered this SDK, otherwise, you risk losing useful info"

File: src/collections/_documentation/platforms/cocoa/usage.md [hunk 12]
```diff
@@ -0,0 +1,51 @@
+---
+title: 'Advanced Usage'
+---
+
+## Capturing uncaught exceptions on macOS
+
+By default macOS application do not crash whenever an uncaught exception occurs. There are some additional steps you have to take to make this work with Sentry. You have to open the applications `Info.plist` file and look for `Principal class`. The content of it which should be `NSApplication` should be replaced with `SentryCrashExceptionApplication`.
```
Comment (PeloWriter) [inline src/collections/_documentation/platforms/cocoa/usage.md]: Can we clarify the steps more specifically here? Suggest:

By default, MacOS applications do not crash whenever an uncaught exception occurs. To enable this with Sentry, 
1. Open the application's `Info.plist` file
2. Search for `Principal class` (the entry is expected to be `NSApplication`)
3. Replace `NSApplication` with `SentryCrashExceptionApplication`

File: src/collections/_documentation/platforms/cocoa/usage.md [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+
+## Dealing with Scopes
+
+There are four different ways to interact with Scopes in the SDK.
```
Comment (PeloWriter) [inline src/collections/_documentation/platforms/cocoa/usage.md]: Can we list the four here and link to their content?

File: src/collections/_documentation/platforms/cocoa/usage.md [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+
+### configureScope
+
+Using `SentrySDK:configureScope` lets you set context data globally, this will be attached to all future events.
```
Comment (PeloWriter) [inline src/collections/_documentation/platforms/cocoa/usage.md]: Suggest the second phrase read, "which will be attached to all future events."

File: src/collections/_documentation/platforms/cocoa/usage.md [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+
+{% include components/platform_content.html content_dir='configureScope' %}
+
+### Passing a Scope Instance
+
+By passing an instance of a Scope directly, it will not merge with the current global Scope. 
+This is helpful in cases where you completly want to control what should be attache to the event.
```
Comment (PeloWriter) [inline src/collections/_documentation/platforms/cocoa/usage.md]: Suggest "Setting an instance of Scope is helpful when you want to completely control what should be attached to the event."

File: src/collections/_documentation/platforms/cocoa/usage.md [hunk 16]
```diff
@@ -0,0 +1,51 @@
...
+
+{% include components/platform_content.html content_dir='pass-scope' %}
+
+### Using Scope Callback
+
+In cases where you might want to keep the global state but want to mutate context data just for one capture call, you might want to consider using the Scope callack:
```
Comment (PeloWriter) [inline src/collections/_documentation/platforms/cocoa/usage.md]: Suggest: "To maintain global state, but mutate context data for one capture call, use the Scope callback:"

File: src/collections/_documentation/platforms/cocoa/usage.md [hunk 17]
```diff
@@ -0,0 +1,51 @@
...
+
+{% include components/platform_content.html content_dir='scope-callback' %}
+
+## Integrations
+
+The SDK comes with a few default integrations. All of them are enabled by default, if you decide to disable any of them you can do it like this:
```
Comment (PeloWriter) [inline src/collections/_documentation/platforms/cocoa/usage.md]: Suggest: "The SDK enables all integrations by default. To disable any of them:"

File: src/collections/_documentation/platforms/cocoa/usage.md [hunk 18]
```diff
@@ -0,0 +1,51 @@
...
+
+{% include components/platform_content.html content_dir='integrations' %}
+
+#### SentryCrashIntegration
+
+This integration is the core part of the SDK. It hooks into all signal and exception handlers to capture uncaught errors or crashes. This integration is also responsible for adding most of the device information to the events so beware of when disabling this, you will not get crash reports nor will your events contain a lot of device data.
```
Comment (PeloWriter) [inline src/collections/_documentation/platforms/cocoa/usage.md]: Suggest splitting the final sentence into two so it's easier to parse. "This integration is also responsible for adding most of the device information to events. If it is disabled, you will not receive crash reports, nor will events contain much device data."

File: src/collections/_documentation/platforms/cocoa/usage.md [hunk 19]
```diff
@@ -0,0 +1,51 @@
...
+
+#### SentryUIKitMemoryWarningIntegration
+
+This integration will only do something in app where UIKit is available (everywhere else it's noop). It will send an event to Sentry when `UIApplicationDidReceiveMemoryWarningNotification` is received.
```
Comment (PeloWriter) [inline src/collections/_documentation/platforms/cocoa/usage.md]: Suggest: "This integration only functions in apps where UIKit is available. The integration will send..."

File: src/collections/_documentation/clients/cocoa/index.md [hunk 20]
```diff
@@ -1,6 +1,6 @@
 ---
 title: Cocoa
-sidebar_relocation: platforms
+robots: noindex
 ---
 
 This is the documentation for our official clients for Cocoa (Swift and Objective-C). Starting with version `3.0.0` we’ve switched our internal code from Swift to Objective-C to maximize compatibility. Also we trimmed the public API of our sdk to a minimum. Some of the lesser used features that were present before are gone now, check out [Migration Guide]({%- link _documentation/clients/cocoa/migration.md -%}#migration) or [Advanced Usage]({%- link _documentation/clients/cocoa/advanced.md -%}#advanced) for details.
```
[the remark this answers is not in the document]
Comment (HazAT) [reply src/collections/_documentation/clients/cocoa/index.md]: So this is actually the old page, old docs 😅 
But nevertheless, I updated the paragraph.

File: src/collections/_documentation/platforms/cocoa/index.md (hunk 4 above)
Comment (HazAT) [reply src/collections/_documentation/platforms/cocoa/index.md]: I removed the whole paragraph.

**Comment by HazAT** [comment]
Thanks @PeloWriter for the detailed review :)

**Comment by HazAT** [comment]
@fdrabek @PeloWriter @MimiDumpling 
Can I merge this :)?

**Comment by MimiDumpling** [comment]
> @fdrabek @PeloWriter @MimiDumpling
> Can I merge this :)?

@PeloWriter did the review. I think she should have final approval from the Docs side. :)
